apps/fyle/tasks.py

```
@@ -138,6 +138,16 @@ def import_expenses(workspace_id, accounting_export: AccountingExport = None, so
                 if imported_from != ExpenseImportSourceEnum.CONFIGURATION_UPDATE:
                     workspace.save()
 
+            # Handle fund source changes for state change events (report state changes)
+            if is_state_change_event and report_id:
+                try:
+                    expenses_count = Expense.objects.filter(workspace_id=workspace_id, report_id=report_id).count()
+                    if expenses_count > 0 and report_state in ('APPROVED', 'ADMIN_APPROVED'):
+                        logger.info("Handling expense fund source change for workspace_id: %s, report_id: %s", workspace_id, report_id)
+                        handle_expense_fund_source_change(workspace_id, report_id, platform)
+                except Exception as e:
+                    logger.exception("Error handling expense fund source change for workspace_id: %s, report_id: %s | ERROR: %s", workspace_id, report_id, e)
```

Ruff (0.12.2)

Redundant exception object included in `logging.exception` call

(TRY401)